src/components/app/details/triggerView/ApprovalNode/ApprovalMaterial.tsx

```diff
@@ -0,0 +1,512 @@
+import { submitApprovalRequest } from './Service'
+import { TriggerViewContext } from '../config'
+import Tippy from '@tippyjs/react'
+import { toast } from 'react-toastify'
+import { ApprovedTippyContent } from './ApprovalMaterial.component'
+import {
+    APPROVAL_ACTION_TYPE,
+    APPROVAL_CTA_TEXT,
+    APPROVAL_REQUEST_TOAST_MSG,
+    APPROVAL_RUNTIME_STATE,
+} from './Constants'
+import { ApprovalRequestType } from './Types'
+
+export default function ApprovalMaterial({
+    material,
+    envName,
+    stageType,
+    changeTab,
+    toggleSourceInfo,
+    appId,
+    pipelineId,
+    parentEnvironmentName,
+    node,
+    selectedTabIndex,
+}) {
+    const { onClickCDMaterial } = useContext(TriggerViewContext)
+    const [tippyVisible, setTippyVisible] = useState<Record<string, boolean>>({})
+    const [requestInProgress, setRequestInProgress] = useState(false)
+    const loginInfo = getLoginInfo()
+    const email: string = loginInfo ? loginInfo['email'] || loginInfo['sub'] : ''
+    const [, securityModuleRes] = useAsync(() => getModuleInfo(ModuleNameMap.SECURITY), [])
+    const isSecurityModuleInstalled = securityModuleRes?.status === ModuleStatus.INSTALLED
+
+    const renderActiveEnv = (envName: string) => {
+        return (
+            <span className="bcg-1 br-4 eg-2 cn-9 pt-3 pb-3 pl-6 pr-6 bw-1 mr-6">
+                <div className="fw-4 fs-11 lh-16 flex">
+                    <World className="icon-dim-16 mr-4 scg-5" />
+                    Active on <span className="fw-6 ml-4">{envName} </span>
+                </div>
+            </span>
+        )
+    }
+
+    const renderActiveCD = (mat) => {
+        return (
+            <>
+                {mat.latest && renderActiveEnv(envName)}
+                {mat.runningOnParentCd && renderActiveEnv(parentEnvironmentName)}
+            </>
+        )
+    }
+
+    const renderFailedCD = () => {
+        return (
+            <span className="bcr-1 br-4 er-2 cn-9 pt-3 pb-3 pl-6 pr-6 bw-1 mr-6">
+                <div className="fw-4 fs-11 lh-16 flex">
+                    <Failed className="icon-dim-16 mr-4" />
+                    Last deployment failed on <span className="fw-6 ml-4">{envName} </span>
+                </div>
+            </span>
+        )
+    }
+
+    const renderProgressingCD = () => {
+        return (
+            <span className="bcy-1 br-4 ey-2 cn-9 pt-3 pb-3 pl-6 pr-6 bw-1 mr-6">
+                <div className="fw-4 fs-11 lh-16 flex">
+                    <div className={`dc__app-summary__icon icon-dim-16 mr-6 progressing progressing--node`}></div>
+                    Deploying on <span className="fw-6 ml-4">{envName} </span>
+                </div>
+            </span>
+        )
+    }
+
+    const renderSequentialCDCardTitle = (mat) => {
+        if (
+            mat.latest ||
+            mat.runningOnParentCd ||
+            mat.artifactStatus === 'Progressing' ||
```

Comment:
we should move these statuses in some const

Reply:
Done